RuntimeException if hue bridge cannot be accessed due to network issues

I found this in my logs:
```
14:58:31.884 [ERROR] [nternal.profiles.DefaultMasterProfile] - Exception occurred while calling handler: java.lang.RuntimeException: java.net.ConnectException: Network is unreachable
java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.net.ConnectException: Network is unreachable
	at org.eclipse.smarthome.core.common.SafeMethodCaller.executeDirectly(SafeMethodCaller.java:220) [99:org.eclipse.smarthome.core:0.9.0.201709260841]
	at org.eclipse.smarthome.core.common.SafeMethodCaller.callAsynchronous(SafeMethodCaller.java:189) [99:org.eclipse.smarthome.core:0.9.0.201709260841]
	at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:83) [99:org.eclipse.smarthome.core:0.9.0.201709260841]
	at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:67) [99:org.eclipse.smarthome.core:0.9.0.201709260841]
	at org.eclipse.smarthome.core.thing.internal.profiles.DefaultMasterProfile.onCommand(DefaultMasterProfile.java:49) [106:org.eclipse.smarthome.core.thing:0.9.0.201709260841]
	at org.eclipse.smarthome.core.thing.internal.CommunicationManager.lambda$2(CommunicationManager.java:177) [106:org.eclipse.smarthome.core.thing:0.9.0.201709260841]
	at org.eclipse.smarthome.core.thing.internal.CommunicationManager$$Lambda$378/1888240791.accept(Unknown Source) [106:org.eclipse.smarthome.core.thing:0.9.0.201709260841]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) [?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175) [?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175) [?:?]
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948) [?:?]
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580) [?:?]
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:270) [?:?]
	at java.util.concurrent.ConcurrentHashMap$ValueSpliterator.forEachRemaining(ConcurrentHashMap.java:3566) [?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:512) [?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502) [?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) [?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) [?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) [?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418) [?:?]
	at org.eclipse.smarthome.core.thing.internal.CommunicationManager.receiveCommand(CommunicationManager.java:172) [106:org.eclipse.smarthome.core.thing:0.9.0.201709260841]
	at org.eclipse.smarthome.core.thing.internal.CommunicationManager.receive(CommunicationManager.java:91) [106:org.eclipse.smarthome.core.thing:0.9.0.201709260841]
	at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:207) [99:org.eclipse.smarthome.core:0.9.0.201709260841]
	at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:1) [99:org.eclipse.smarthome.core:0.9.0.201709260841]
	at org.eclipse.smarthome.core.common.SafeMethodCaller$CallableWrapper.call(SafeMethodCaller.java:181) [99:org.eclipse.smarthome.core:0.9.0.201709260841]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:?]
	at java.lang.Thread.run(Thread.java:745) [?:?]
Caused by: java.lang.RuntimeException: java.net.ConnectException: Network is unreachable
	at org.eclipse.smarthome.binding.hue.handler.HueBridgeHandler.updateLightState(HueBridgeHandler.java:198) ~[?:?]
	at org.eclipse.smarthome.binding.hue.handler.HueLightHandler.handleCommand(HueLightHandler.java:267) ~[?:?]
	at org.eclipse.smarthome.core.thing.internal.profiles.DefaultMasterProfile$1.call(DefaultMasterProfile.java:52) ~[?:?]
	at org.eclipse.smarthome.core.thing.internal.profiles.DefaultMasterProfile$1.call(DefaultMasterProfile.java:1) ~[?:?]
	at org.eclipse.smarthome.core.common.SafeMethodCaller.executeDirectly(SafeMethodCaller.java:218) ~[?:?]
	... 28 more
```

--- a/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueBridgeHandler.java
+++ b/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueBridgeHandler.java
@@ -187,15 +187,17 @@ public class HueBridgeHandler extends ConfigStatusBridgeHandler {
     }
 
     public void updateLightState(FullLight light, StateUpdate stateUpdate) {
-
         if (hueBridge != null) {
             try {
                 hueBridge.setLightState(light, stateUpdate);
             } catch (DeviceOffException e) {
                 updateLightState(light, LightStateConverter.toOnOffLightState(OnOffType.ON));
                 updateLightState(light, stateUpdate);
-            } catch (IOException | ApiException e) {
-                throw new RuntimeException(e);
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            } catch (ApiException e) {
+                // This should not happen - if it does, it is most likely some bug that should be reported.
+                logger.warn("Error while accessing light: {}", e.getMessage(), e);
             } catch (IllegalStateException e) {
                 logger.trace("Error while accessing light: {}", e.getMessage());
             }

--- a/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java
+++ b/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java
@@ -454,12 +454,8 @@ public class HueLightHandler extends BaseThingHandler implements LightStatusList
         int delay = getAlertDuration(command);
 
         if (delay > 0) {
-            scheduledFuture = scheduler.schedule(new Runnable() {
-
-                @Override
-                public void run() {
-                    updateState(CHANNEL_ALERT, new StringType(LightStateConverter.ALERT_MODE_NONE));
-                }
+            scheduledFuture = scheduler.schedule(() -> {
+                updateState(CHANNEL_ALERT, new StringType(LightStateConverter.ALERT_MODE_NONE));
             }, delay, TimeUnit.MILLISECONDS);
         }
     }

--- a/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/HueBridge.java
+++ b/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/HueBridge.java
@@ -248,13 +248,13 @@ public class HueBridge {
      * @throws UnauthorizedException thrown if the user no longer exists
      * @throws EntityNotAvailableException thrown if the specified light no longer exists
      * @throws DeviceOffException thrown if the specified light is turned off
+     * @throws IOException if the bridge cannot be reached
      */
     public void setLightState(Light light, StateUpdate update) throws IOException, ApiException {
         requireAuthentication();
 
         String body = update.toJson();
         Result result = http.put(getRelativeURL("lights/" + enc(light.getId()) + "/state"), body);
-
         handleErrors(result);
     }
 

